src/Diffusion.jl

```diff
@@ -1,9 +1,76 @@
 module Diffusion
+using StaticArrays 
 
 export absorb_boundry
+export reflect_boundary 
+export slope
+export midpoint
+export circleintersect
+export rslope
+export perpslope
+export solve
+export posFinal
 
 function absorb_boundry(pos,r)
-    return pos + r
+    if(((pos[1]^2)+(pos[2]^2))^(1/2) < r)
+    return pos
+    else
+    return nothing
+    end
+end
+
+function slope(pos1,pos2)               #finds the slope between two points
+    return (pos1[2]-pos2[2])/(pos1[1]-pos2[1])
```

Comment:
You could also implement this as `rslope(pos2 - pos1)`. If `pos1` and `pos2` are StaticVectors, the generated code will be the same as with your expression (all "object creation" gets removed by the compiler).